ksql/api.py

```diff
@@ -141,8 +138,8 @@ def _request2(self, endpoint, connection, body, method="POST", encoding="utf-8")
             base64string = base64.b64encode(bytes("{}:{}".format(self.api_key, self.secret), "utf-8")).decode("utf-8")
             headers["Authorization"] = "Basic %s" % base64string
 
-        connection.request(method=method.upper(), url=url, headers=headers, body=data)
-        resp = connection.get_response()
+        resp = connection(method=method.upper(), url=url, headers=headers, body=data)
```

Comment:
Sorry for my 5 cents, but I tried testing queries with HTTP/2 and found an issue which leads here. `Connection` (which is `httpx.request` object here) does not want `body`, but `data` argument instead. And it wants a dict object for `data`, not a binary chunk. It looks like more refactoring is in order.